### PR TITLE
#10034: Include SortyBy parameter in the CSW catalog query

### PR DESCRIPTION
**Pull request by dsuren1**
## Description
This PR adds a new field to CSW catalog's advanced setting which adds `SortBy` to csw query

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Feature

## Issue

**What is the current behavior?**
- #10034 

**What is the new behavior?**
- Adding name to sortby field adds `SortBy` to catalog query
- By default sort order is ASC

### Screenshot
image

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Review by MV88 — CHANGES_REQUESTED** [review]
a few things here and there

also can you create a map where the service has a working filter and link it here?

i tried with dc:title using our gs-stable but it was not working when saving the map

here is the req:
```
curl 'https://gs-stable.geo-solutions.it/geoserver/csw?service=CSW&version=2.0.2&request=' \
  -H 'authority: gs-stable.geo-solutions.it' \
  -H 'accept: application/json, text/plain, */*' \
  -H 'accept-language: en-US,en;q=0.9' \
  -H 'cache-control: no-cache' \
  -H 'content-type: application/xml' \
  -H 'origin: http://localhost:8081' \
  -H 'pragma: no-cache' \
  -H 'referer: http://localhost:8081/' \
  -H 'sec-ch-ua: "Chromium";v="122", "Not(A:Brand";v="24", "Google Chrome";v="122"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Windows"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: cross-site' \
  -H 'user-agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36' \
  --data-raw $'<csw:GetRecords xmlns:csw="http://www.opengis.net/cat/csw/2.0.2" xmlns:ogc="http://www.opengis.net/ogc" xmlns:gml="http://www.opengis.net/gml" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:dct="http://purl.org/dc/terms/" xmlns:gmd="http://www.isotc211.org/2005/gmd" xmlns:gco="http://www.isotc211.org/2005/gco" xmlns:gmi="http://www.isotc211.org/2005/gmi" xmlns:ows="http://www.opengis.net/ows" service="CSW" version="2.0.2" resultType="results" startPosition="1" maxRecords="4"> <csw:Query typeNames="csw:Record"> <csw:ElementSetName>full</csw:ElementSetName> <csw:Constraint version="1.1.0"> <ogc:Filter> <ogc:Or>\n    <ogc:PropertyIsEqualTo>\n        <ogc:PropertyName>dc:type</ogc:PropertyName>\n        <ogc:Literal>dataset</ogc:Literal>\n    </ogc:PropertyIsEqualTo>\n    <ogc:PropertyIsEqualTo>\n        <ogc:PropertyName>dc:type</ogc:PropertyName>\n        <ogc:Literal>http://purl.org/dc/dcmitype/Dataset</ogc:Literal>\n    </ogc:PropertyIsEqualTo>\n</ogc:Or> </ogc:Filter> </csw:Constraint> <ogc:SortBy><ogc:SortProperty><ogc:PropertyName>dc:title</ogc:PropertyName><ogc:SortOrder>ASC</ogc:SortOrder></ogc:SortProperty></ogc:SortBy> </csw:Query> </csw:GetRecords>'
```

File: web/client/api/CSW.js [hunk 1]
```diff
@@ -74,13 +82,15 @@ export const cswGetRecordsXml = '<csw:GetRecords xmlns:csw="http://www.opengis.n
...
+export const constructXMLBody = (startPosition, maxRecords, searchText, { options: { service } = {} } = {}) => {
+    const { filter, sortBy: sortObj } = service ?? {};
     const staticFilter = filter?.staticFilter || defaultStaticFilter;
     const dynamicFilter = `<ogc:And>
         ${template(filter?.dynamicFilter || defaultDynamicFilter)({ searchText })}
         ${staticFilter}
     </ogc:And>`;
-    return template(cswGetRecordsXml)({ filterXml: !searchText ? staticFilter : dynamicFilter, startPosition, maxRecords });
+    const sortExp = sortObj?.name ? template(sortBy)({ name: sortObj?.name, order: sortObj?.order ?? "ASC"}) : '';
+    return template(cswGetRecordsXml)({ filterXml: !searchText ? staticFilter : dynamicFilter, startPosition, maxRecords, sortBy: sortExp});
```
[the remark this answers is not in the document]
Comment (dsuren1) [reply web/client/api/CSW.js:93]: A variable with a similar name is already in scope [sortBy](https://github.com/geosolutions-it/MapStore2/pull/10055/files#diff-3a386a6dba2b51b665c13451e244211a772223b7d40627d2a580aff473ef2cd9R47)

**Review by MV88 — APPROVED** [review]

**Comment by MV88** [comment]
@ElenaGallo please test it in DEV

**Comment by ElenaGallo** [comment]
When the Sortyby option is not available (like for gs-stable csw) that option should be disabled in the UI or should the error message be more specific? @tdipisa @dsuren1 what do you think ?

https://github.com/geosolutions-it/MapStore2/assets/56537133/c23dbf4c-90ca-4279-84f7-7d231884996c

**Comment by dsuren1** [comment]
As reported [here](https://github.com/geosolutions-it/MapStore2/issues/10034#issuecomment-1997520424), the service handles it differently. Meaning some time there is error thrown but in other cases it is simply ignored. There isn't a definite way to say the error is caused by sort if we rely on error message in the response

**Pull request merged**